### ISSUE TITLE
Format `Market Share Target` and `SDA` articles in consistent template

### ISSUE THREAD

**Issue by jdhoffa** [RMI-PACTA/r2dii.analysis#457]
See examples of the articles here:
https://rmi-pacta.github.io/r2dii.analysis/articles/target-market-share.html
https://rmi-pacta.github.io/r2dii.analysis/articles/target-sda.html

We (Nick and I) have been thinking that it may be a good idea to define our own "template" for "RMI/ PACTA metrics". 
This template could include sections around:
* Audience
* Description/ Formulae
* Expected outcome
* Impactful actions: e.g. "Use in a dialogue with a company"
* Use in combination with other metrics
* etc. 

This would help focus and align our metrics, and give us a common starting point towards developing new metrics. 

Obviously the scope of this is broader than this particular package, but I am thinking it may be good to prototype it here as Nick and I work out the kinks. 

[AB#9900](https://dev.azure.com/RMI-PACTA/d77fc8f8-abac-42ae-9e31-846f9901290d/_workitems/edit/9900)

**Comment by jdhoffa** [comment]
I have had a chance to brainstorm a metric template here: 
[On metrics](https://rockmtnins.sharepoint.com/:fl:/g/contentstorage/CSP_28c7e85b-57c6-4b29-ac79-082d6de92b5a/EdD2KPrbGeFEvGUKht6z83kBw8GfMa6mg1qZgW-Np-MbWw?e=3K69zD&nav=cz0lMkZjb250ZW50c3RvcmFnZSUyRkNTUF8yOGM3ZTg1Yi01N2M2LTRiMjktYWM3OS0wODJkNmRlOTJiNWEmZD1iJTIxVy1qSEtNWlhLVXVzZVFndGJla3JXZ1lIWTQ4Mk16NU5oZ3ZpcnFCTnNkdDNpRmZuVWtGRFFMd1FrRjdhbTJmcCZmPTAxQjdDRUtLV1E2WVVQVldZWjRGQ0xZWklLUTNQTEg0M1omYz0lMkYmYT1Mb29wQXBwJnA9JTQwZmx1aWR4JTJGbG9vcC1wYWdlLWNvbnRhaW5lciZ4PSU3QiUyMnclMjIlM0ElMjJUMFJUVUh4eWIyTnJiWFJ1YVc1ekxuTm9ZWEpsY0c5cGJuUXVZMjl0ZkdJaFZ5MXFTRXROV2xoTFZYVnpaVkZuZEdKbGEzSlhaMWxJV1RRNE1rMTZOVTVvWjNacGNuRkNUbk5rZEROcFJtWnVWV3RHUkZGTWQxRnJSamRoYlRKbWNId3dNVUkzUTBWTFMxZExOMWRYVkZGQldFMUdXa0ZKVFV4T1Z6SlBRVlpHVVZoQyUyMiUyQyUyMmklMjIlM0ElMjI1Y2I0Y2RmZi01MmU4LTRmYTEtODk4Yi1lMzU0MWY1ZGQ5YmUlMjIlN0Q%3D)

**Comment by jdhoffa** [comment]
Relates to #320

**Comment by MonikaFu** [comment]
Sounds like a very good idea to me! Maybe we could also link it to the plots we use for specific metrics where applicable. (I also proposed that in the loop document, not sure where the discussion is now.

**Comment by jdhoffa** [comment]
Definitely. I wonder also if it may be preferable to have it in a top-level wrapper package (e.g. pactaverse) or something like that, so it can reference any plot or dataset necessary?

**Comment by MonikaFu** [comment]
Makes sense about the top level. Would we then move all documentation to the top level? Or would we maintain both?

Also (on the other hand), I am using `r2dii.data` and `r2dii.analysis` in the documentation for `r2dii.plot` so maybe vice versa is not such a bad idea. When you render the documentation, it would expose (some) breaking changes to `r2dii.plot`.

**Comment by jdhoffa** [comment]
My gut feeling is that having crisscrossed dependencies management can get quite confusing to manage.
e.g. managing APIs in the direction:
`r2dii.data`-> `r2dii.match` -> `r2dii.analysis` -> `r2dii.plot`

is quite simple. If anything earlier in the chain makes a "breaking API change" it's their responsibility to communicate it well, and everyone downstream to adapt. 

When there is a bidirectional arrow on any of those, it gets very confusing to know who's responsibility it is to manage what.

Same as if `dplyr` makes a breaking API change, they notify us and we change things. But `dplyr` doesn't check a prior if changes they make are going to break `r2dii.analysis`

**Comment by jacobvjk** [comment]
Tell me if it is beyond the scope of the discussion, but in general I think we dearly need a well maintained public place where we have our full methodological documentation that we can cite/link to on reports. We have consistently been rewording the same pieces of text a slight bit to avoid "plagiarising" something that we came up with ourselves, because there was nothing available that we could properly cite. I don't know if everything needs to be in the same place. E.g. modelling of market share / SDA may be better documented in GitHub, whereas the scenario reference document may not. But I think this is something to consider, especially when trying to avoid documenting the same thing in multiple places.

**Comment by jdhoffa** [comment]
@jacobvjk I see this issue as a very clear first step to achieving this, yes.

**Comment by jacobvjk** [comment]
sounds like a good topic to me. Maybe not even controversial, but still good to have a short group discussion about it

**Comment by MonikaFu** [comment]
> My gut feeling is that having crisscrossed dependencies management can get quite confusing to manage. e.g. managing APIs in the direction: `r2dii.data`-> `r2dii.match` -> `r2dii.analysis` -> `r2dii.plot`
> 
> is quite simple. If anything earlier in the chain makes a "breaking API change" it's their responsibility to communicate it well, and everyone downstream to adapt.
> 
> When there is a bidirectional arrow on any of those, it gets very confusing to know who's responsibility it is to manage what.
> 
> Same as if `dplyr` makes a breaking API change, they notify us and we change things. But `dplyr` doesn't check a prior if changes they make are going to break `r2dii.analysis`

I understand and in principle, I agree. However, with `r2dii` suite they are very tightly bound and I think that it might be useful to see the metrics visualized right next to the mathematical explanation of how they work. If that were the case then the need for upstream dependency management would be a side product.

**Comment by AlexAxthelm** [comment]
> However, with r2dii suite they are very tightly bound and I think that it might be useful to see the metrics visualized right next to the mathematical explanation of how they work.

I think that's exactly for [`Suggests`](https://r-pkgs.org/description.html#sec-description-imports-suggests) dependencies, rather than `Imports` (soft vs hard deps) We can put the downstream packages (like `.plot`) as `Suggests` dependencies, which means that we avoid a circular dependency problem, but can still use it while rendering documentation. Then the cocs for the methodology can live next to the methodology

**Comment by jdhoffa** [comment]
Understood, I still forsee potential issues if the mental model is circular in that sense, but definitely happy to be convinced otherwise!

**Comment by jacobvjk** [comment]
Another idea for combining everything into its full glory would be to create cook books for different use cases (which we are planning on doing in any case). I think this would probably not go into the level of detail that we would want to have in the individual repo documentation, but I think it is a great place for clarifying how a given metric is generated, visualised, and interpreted. There will probably be overlaps with the metric template, but it gives us space to avoid cramming all possible information into one document.

**Comment by jdhoffa** [comment]
Update comment, but yeah that's exactly what I had in mind really. 
`pactaverse` could contain many many different vignettes, that walk through lots of different potential use-cases using demo data. 

It could simultaneously be a demonstrative library of all of our metrics and packages AND an automated CI/CD system 

@AlexAxthelm I imagine we could build an Action that ensures `pactaverse` builds all vignettes on any push to `main` across any of our packages. That could be a good "universal" check of the API consistency across all packages.

**Comment by AlexAxthelm** [comment]
> If the goal is to have demo visuals and data living under the same roof, I think that place would be an umbrella package (e.g. https://github.com/RMI-PACTA/pactaverse)

I think this can lead to complications, since you would need to coordinate PRs across repos. It would be better to have 1 pr that has code and docs changes together, and depend on optional packages for generating docs _where they are implemented_. So methodology docs would likve in `.analysis`, plots about visualizations in `.plot`, etc.)

If you look at the `tidyverse` vignettes (https://github.com/tidyverse/tidyverse/tree/main/vignettes), they are generally about the "big picture" of using the whole ecosystem together.

https://r-pkgs.org/dependencies-in-practice.html#sec-dependencies-in-suggests-in-examples-and-vignettes

**Comment by jdhoffa** [comment]
^that's exactly what I am talking about

**Comment by cjyetman** [comment]
[pacta.loanbook](https://github.com/RMI-PACTA/pacta.loanbook) seems to integrate some of the suggested solutions here, though not the broad CI/CD testing of the entire suite of packages. @jdhoffa and @jacobvjk is this something we should be considering for pacta.loanbook (also, feel free to move the discussion over to pacta.loanbook if so)?

**Comment by jacobvjk** [comment]
> [pacta.loanbook](https://github.com/RMI-PACTA/pacta.loanbook) seems to integrate some of the suggested solutions here, though not the broad CI/CD testing of the entire suite of packages. [@jdhoffa](https://github.com/jdhoffa) and [@jacobvjk](https://github.com/jacobvjk) is this something we should be considering for pacta.loanbook (also, feel free to move the discussion over to pacta.loanbook if so)?

Seems like a good idea to pick up this conversation in the [pacta.loanbook](https://github.com/RMI-PACTA/pacta.loanbook) repo again. Our initial feedback seems to highlight a need for this type of documentation, so this may be time to get it done

**Comment by cjyetman** [comment]
@jdhoffa and @jacobvjk can you distill the relevant, discrete ideas here and [make separate issues for them in pacta.loanbook](https://github.com/RMI-PACTA/pacta.loanbook/issues/new?template=Blank+issue) so they're hard to forget and easier to tackle?

**Comment by jdhoffa** [comment]
Superseded by:
https://github.com/RMI-PACTA/pacta.loanbook/issues/93
and
https://github.com/RMI-PACTA/pacta.loanbook/issues/94
